src/main/java/com/minecolonies/structures/client/TemplateTessellator.java

```diff
@@ -0,0 +1,162 @@
+package com.minecolonies.structures.client;
+
+import com.minecolonies.structures.lib.RenderUtil;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.*;
+import net.minecraft.client.renderer.texture.ITextureObject;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexBuffer;
+import net.minecraft.client.renderer.vertex.VertexFormatElement;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.Rotation;
+import net.minecraft.util.math.BlockPos;
+
+import static org.lwjgl.opengl.GL11.*;
+
+public class TemplateTessellator
+{
+
+    private static final int VERTEX_COMPONENT_SIZE = 3;
+    private static final  int   COLOR_COMPONENT_SIZE              = 4;
+    private static final  int   TEX_COORD_COMPONENT_SIZE          = 2;
+    private static final  int   LIGHT_TEX_COORD_COMPONENT_SIZE    = TEX_COORD_COMPONENT_SIZE;
+    private static final  int   VERTEX_SIZE                       = 28;
+    private static final  int   VERTEX_COMPONENT_OFFSET           = 0;
+    private static final  int   COLOR_COMPONENT_OFFSET            = 12;
+    private static final  int   TEX_COORD_COMPONENT_OFFSET        = 16;
+    private static final  int   LIGHT_TEXT_COORD_COMPONENT_OFFSET = 24;
+    private static final float HALF_PERCENT_SHRINK               = 0.995F;
+    private static final int DEFAULT_BUFFER_SIZE = 2097152;
+
+    private final BufferBuilder builder;
+    private final VertexBuffer         buffer      = new VertexBuffer(DefaultVertexFormats.BLOCK);
+    private final VertexBufferUploader vboUploader = new VertexBufferUploader();
+    private       boolean              isReadOnly  = false;
+
+    public TemplateTessellator()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 2097152 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add underscores to this numeric value for readability [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2148)